metadata-builders/src/main/java/com/linkedin/metadata/builders/search/MLModelIndexBuilder.java

```diff
@@ -0,0 +1,156 @@
+import com.linkedin.metadata.search.MLModelDocument;
+import com.linkedin.metadata.snapshot.MLModelSnapshot;
+import com.linkedin.ml.metadata.BaseData;
+import com.linkedin.ml.metadata.EvaluationData;
+import com.linkedin.ml.metadata.MLModelProperties;
+import com.linkedin.ml.metadata.TrainingData;
+
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MLModelIndexBuilder extends BaseIndexBuilder<MLModelDocument> {
+
+    public MLModelIndexBuilder() {
+        super(Collections.singletonList(MLModelSnapshot.class), MLModelDocument.class);
+    }
+
+    @Nonnull
+    private static String buildBrowsePath(@Nonnull MLModelUrn urn) {
+        return ("/" + urn.getOriginEntity() + "/"  + urn.getPlatformEntity().getPlatformNameEntity() + "/" + urn.getMlModelNameEntity())
+            .replace('.', '/').toLowerCase();
+    }
+
+    /**
+     * Given model urn, this returns a {@link MLModelDocument} model that has urn, dataset name, platform and origin fields set
+     *
+     * @param urn {@link MLModelUrn} that needs to be set
+     * @return {@link MLModelDocument} model with relevant fields set that are extracted from the urn
+     */
+    @Nonnull
+    private static MLModelDocument setUrnDerivedFields(@Nonnull MLModelUrn urn) {
+        return new MLModelDocument()
+            .setName(urn.getMlModelNameEntity())
+            .setOrigin(urn.getOriginEntity())
+            .setPlatform(urn.getPlatformEntity().getPlatformNameEntity())
+            .setUrn(urn);
+    }
+
+    @Nonnull
+    private List<MLModelDocument> getDocumentsToUpdateFromSnapshotType(@Nonnull MLModelSnapshot mlModelSnapshot) {
+        final MLModelUrn urn = mlModelSnapshot.getUrn();
+        final List<MLModelDocument> documents =  mlModelSnapshot.getAspects().stream().map(aspect -> {
+            if (aspect.isDeprecation()) {
+                return getDocumentToUpdateFromAspect(urn, aspect.getDeprecation());
+            } else if (aspect.isEvaluationData()) {
+                return getDocumentToUpdateFromAspect(urn, aspect.getEvaluationData());
+            } else if (aspect.isMLModelProperties()) {
+                return getDocumentToUpdateFromAspect(urn, aspect.getMLModelProperties());
+            } else if (aspect.isOwnership()) {
+                return getDocumentToUpdateFromAspect(urn, aspect.getOwnership());
+            } else if (aspect.isStatus()) {
+                return getDocumentToUpdateFromAspect(urn, aspect.getStatus());
+            } else if (aspect.isTrainingData()) {
+                return getDocumentToUpdateFromAspect(urn, aspect.getTrainingData());
+            }
+            return null;
+        }).filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        documents.add(setUrnDerivedFields(urn));
+        return documents;
+    }
+
+    @Nonnull
+    private MLModelDocument getDocumentToUpdateFromAspect(MLModelUrn urn, Deprecation deprecation) {
+        return new MLModelDocument()
```

Comment:
Sorry for the confusion. Looks like we will need to set the `urn` field (not necessarily the urn parts) for each document. You can refer to this PR for more info https://github.com/linkedin/datahub/pull/1972

Reply:
@RyanHolstien : this comment still seems valid?

Reply:
I added the .setUrn() call to each of these methods, did I misunderstand?

Reply:
There might be some github confusion here. If you click this comment thread, you'll view this file at the commit that Jyoti made the commend on. Looks fine at the latest commit?

Reply:
This is addressed. Can be resolved

Reply:
Cool... good to resolve then.